### PR TITLE
[Ecommerce][IndexService] Errorhandling: add subObjectId to error event

### PR DESCRIPTION
**Pull request by jremmurd**
# Enhancement

This PR adds the subObjectId to the PreProcessErrorEvent.

### CODE REVIEW

File: lib/Event/Model/Ecommerce/IndexService/PreprocessErrorEvent.php [hunk 1]
```diff
@@ -27,17 +27,23 @@ class PreprocessErrorEvent extends Event
...
+    protected $subObjectId;
+    
     /**
      * PreprocessErrorEvent constructor.
      *
      * @param \Throwable $exception
      * @param bool $throwException
      */
-    public function __construct(\Throwable $exception, bool $throwException = true)
+    public function __construct(\Throwable $exception, bool $throwException = true, $subObjectId = 0)
```
Comment (weisswurstkanone) [inline lib/Event/Model/Ecommerce/IndexService/PreprocessErrorEvent.php]: please adapt the phodoc

Comment (jdreesen) [reply lib/Event/Model/Ecommerce/IndexService/PreprocessErrorEvent.php]: Should the parameter be typed?
```suggestion
    public function __construct(\Throwable $exception, bool $throwException = true, int $subObjectId = 0)
```

Comment (jremmurd) [reply lib/Event/Model/Ecommerce/IndexService/PreprocessErrorEvent.php]: thanks!

**Comment by fashxp** [comment]
thx!

**Pull request merged**